artview/plugins/cmap.py

```diff
@@ -0,0 +1,307 @@
+        if self.cmap is None:
+            return
+
+        self.ax.cla()
+        self.cax.cla()
+        cmap = self.cmap
+        if 'norm' not in cmap or cmap['norm'] is None:
+            self.norm_type.setCurrentIndex(0)
+        else:
+            norm_name = cmap['norm'].__class__.__name__
+            if norm_name == 'Normalize':
+                self.norm_type.setCurrentIndex(1)
+            elif norm_name == 'LogNorm':
+                self.norm_type.setCurrentIndex(2)
+            elif norm_name == 'SymLogNorm':
+                self.norm_type.setCurrentIndex(3)
+            elif norm_name == 'PowerNorm':
+                self.norm_type.setCurrentIndex(4)
+            elif norm_name == 'BoundaryNorm':
+                self.norm_type.setCurrentIndex(5)
+
+        if cmap is not None:
+            if 'norm' in cmap:
+                norm = cmap['norm']
+            else:
+                norm = None
+            im = self.ax.imshow(gradient, aspect='auto', cmap=cmap['cmap'],
+                                vmin=cmap['vmin'], vmax=cmap['vmax'], norm=norm)
+            plt.colorbar(im, cax=self.cax)
+
+        self.canvas.draw()
+
+    def change_norm(self, idx):
+        self.set_layout()
+        self.update_colormap()
+
+    def update_colormap(self):
+        '''Get colormap from GUI.'''
+        self.cmap['lock'] = self.lock_box.isChecked()
+        idx = self.norm_type.currentIndex()
+
+        self.cmap['vmin'] = float(self.ent_vmin.text())
+        self.cmap['vmax'] = float(self.ent_vmax.text())
+
+        if idx == 0:
+            self.cmap['norm'] = None
+        elif idx == 1:
+            self.cmap['norm'] = colors.Normalize(vmin=self.cmap['vmin'],
+                                                 vmax=self.cmap['vmax'])
+        elif idx == 2:
+            self.cmap['norm'] = colors.LogNorm(vmin=self.cmap['vmin'],
+                                               vmax=self.cmap['vmax'])
+        elif idx == 3:
+            self.cmap['norm'] = colors.SymLogNorm(
+                linthresh=float(self.ent_linthresh.text()),
+                linscale=float(self.ent_linscale.text()),
+                vmin=self.cmap['vmin'],
+                vmax=self.cmap['vmax'])
+        elif idx == 4:
+            self.cmap['norm'] = colors.PowerNorm(
+                gamma=float(self.ent_gamma.text()),
+                vmin=self.cmap['vmin'],
+                vmax=self.cmap['vmax'])
+        elif idx == 5:
+            bounds = self.get_bounds()
+            self.cmap['norm'] = colors.BoundaryNorm(bounds,
+                                                    ncolors=256)
+        self.plot()
+
+    def select_cmap(self, post):
+        '''Select cmap from a list of pyart colormaps.'''
+        cmap = core.common.select_cmap().selection
+        if cmap is not None:
+            self.cmap['cmap'] = cmap
+            self.plot()
+
+    def apply(self):
+        '''Apply changes to shared colormap.'''
+        self.Vcolormap.change(self.cmap)
```

Comment:
I agree with calling `update_colormap` on apply. This may solve the issue with the discrete, It look like in your system it does not allow limits with equal values, I'm not really sure how to avoid that.